### PR TITLE
Force new resource when changing apiVersion or Kind

### PR DESCRIPTION
**Pull request by alexsomesan**
### Description

With this change, the `kubernetes_manifest` resource will force the re-creation of the resource when either `apiVersion` or `kind` attributes change.

### Acceptance tests
- [ ] Have you added an acceptance test for the functionality being added?
- [ ] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
...
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: manifest/provider/plan.go [hunk 1]
```diff
@@ -112,6 +112,11 @@ func (s *RawProviderServer) getFieldManagerConfig(v map[string]tftypes.Value) (s
 func (s *RawProviderServer) PlanResourceChange(ctx context.Context, req *tfprotov5.PlanResourceChangeRequest) (*tfprotov5.PlanResourceChangeResponse, error) {
 	resp := &tfprotov5.PlanResourceChangeResponse{}
 
+	resp.RequiresReplace = append(resp.RequiresReplace,
+		tftypes.NewAttributePath().WithAttributeName("manifest").WithAttributeName("apiVersion"),
```
Comment (jrhouston) [inline manifest/provider/plan.go:116]: Should we also add `metadata.name` and `metadata.namespace` to match the built-in resources?

Comment (alexsomesan) [reply manifest/provider/plan.go:116]: Good point!

**Comment by jrhouston** [comment]
Looks like this is failing because we need to check if the resource is namespaced or not.

File: manifest/provider/plan.go [hunk 2]
```diff
@@ -272,6 +271,21 @@ func (s *RawProviderServer) PlanResourceChange(ctx context.Context, req *tfproto
 		return resp, nil
 	}
 
+	ns, err := IsResourceNamespaced(gvk, rm)
```
Comment (jrhouston) [inline manifest/provider/plan.go:274]: couldn't we do this below, where we already have this logic? https://github.com/hashicorp/terraform-provider-kubernetes/blob/f424b51e9c7f6b1757608e0aad43c73c42d2848b/manifest/provider/plan.go#L56-L57

**Review by jrhouston — APPROVED** [review]

**Pull request merged**